cartographer/mapping/internal/3d/local_trajectory_builder_3d.cc

```diff
@@ -150,26 +154,51 @@ LocalTrajectoryBuilder3D::AddRangeData(
   ++num_accumulated_;
 
   if (num_accumulated_ >= options_.num_accumulated_range_data()) {
+    const common::Time current_sensor_time = synchronized_data.time;
+    common::optional<double> sensor_duration;
```

Comment:
Keep this as a `common::Duration` until the computation of the fraction happens? This way, it would be more obvious that this is a time in seconds.